semcore/add-filter-pattern/src/AddFilterPattern.tsx

```diff
@@ -0,0 +1,276 @@
+import React from 'react';
+import { Flex } from '@semcore/flex-box';
+import Button from '@semcore/button';
+import createComponent, { Component, sstyled, Root } from '@semcore/core';
+import DropdownMenu from '@semcore/dropdown-menu';
+import MathPlusM from '@semcore/icon/MathPlus/m';
+import CloseM from '@semcore/icon/Close/m';
+import FilterPatternType, {
+  AddFilterPatternProps,
+  AddFilterPatternItemProps,
+} from './AddFilterPattern.types';
+import AddFilterPatternSelect from './components/AddFilterPatternSelect';
+import AddFilterPatternSearch from './components/AddFilterPatternSearch';
+import AddFilterPatternDropdown from './components/AddFilterPatternDropdown';
+import { findAllComponents } from '@semcore/utils/lib/findComponent';
+
+type AddFilterPatternDoprdownOption = { label: string; value: string };
+type AddFilterDropdownMenuProps = {
+  options: AddFilterPatternDoprdownOption[];
+  toggleFieldVisibility: (name: string, status?: boolean) => void;
+  visibleFilters: Set<string>;
+};
+
+type FilterData = Record<string, any>;
+type ClearButtonProps = {
+  filterData: FilterData;
+  clearAll: () => void;
+};
+
+type AddFilterPatternState = {
+  visibleFilters: Set<string>;
+  addDropdownItems: AddFilterPatternDoprdownOption[];
+  filterData: FilterData;
+};
+
+const getDefaultAddDropdownItems = (
+  props: AddFilterPatternProps,
+): AddFilterPatternDoprdownOption[] => {
+  const childrenArray: React.ReactElement<AddFilterPatternItemProps>[] = Array.isArray(
+    props.children,
+  )
+    ? props.children
+    : [props.children];
+
+  return childrenArray
+    .flat()
+    .filter(({ props }) => !props.alwaysVisible)
+    .map(({ props }) => {
+      const { name, displayName } = props;
+      const value = name;
+      const label = displayName ?? name;
+
+      return { label, value };
+    });
+};
+
+class RootAddFilterPattern extends Component<AddFilterPatternProps, {}, AddFilterPatternState> {
+  static displayName = 'AddFilterPattern';
+  getDefaultAddDropdownItems: () => AddFilterPatternDoprdownOption[];
+  selectMenuRefs: Map<string, HTMLElement>;
+
+  constructor(props: AddFilterPatternProps) {
+    super(props);
+    this.getDefaultAddDropdownItems = () => getDefaultAddDropdownItems(props);
+    this.selectMenuRefs = new Map();
+
+    this.state = {
+      visibleFilters: new Set(),
+      addDropdownItems: this.getDefaultAddDropdownItems(),
+      filterData: {},
```

Comment:
I think, you can't put it here like this.
You should add `filterData` to props, because I can imagine a case where you need to open preset filters. And because of that, you should show filters not only by always visible...

Reply:
Using filterData as prop now